### DISCUSSION TITLE
What is the relationship between tx_osnr and signal in info.py?

### DISCUSSION THREAD

**Discussion by alittleice** [Telecominfraproject/oopt-gnpy#513] [Q&A] [answered]
In the `info.py` file, there is a confusing point, please help to solve.
Here's the confusing part of the code:
```
signal = full(number_of_channels, signal)
baud_rate = full(number_of_channels, baud_rate)   
roll_off = full(number_of_channels, roll_off)    
slot_width = full(number_of_channels, slot_width) if slot_width is not None else \ ceil((1 + roll_off) * baud_rate / DEFAULT_SLOT_WIDTH_STEP) * DEFAULT_SLOT_WIDTH_STEP
chromatic_dispersion = full(number_of_channels, chromatic_dispersion) 
pmd = full(number_of_channels, pmd) 
pdl = full(number_of_channels, pdl) 
latency = full(number_of_channels, latency) 
nli = zeros(number_of_channels) 
ase = zeros(number_of_channels) 
delta_pdb_per_channel = full(number_of_channels, delta_pdb_per_channel) 
tx_osnr = full(number_of_channels, tx_osnr) 
label = full(number_of_channels, label)
```
What do the two parameters `signal `and `tx_osnr `represent and what is the relationship between them? What parameter needs to be modified when the transmit power is configured?
I hope to get your answers. Thank you very much.

**Comment by AndreaDAmico** [comment, marked as answer]
Dear @alittleice,

The two parameters represent two distinct features of the propagated spectrum and there is not any correlations between the them internally in GNPy. In particular,`signal` is the power of the propagated spectrum, whereas, `tx_osnr` is the OSNR at the transmitter.
In general, the OSNRs at the transmitters depend on the transmitted powers that compose the spectrum signal, but this correlation is not managed into GNPy and is left to the user to input adequate values of these two quantities.
